service-commands/scripts/userClockValues.js

```diff
@@ -0,0 +1,153 @@
+const CreatorNode = require('@audius/libs/src/services/creatorNode')
+const { Command } = require('commander')
+
+function commaSeparatedList(value, unusedPrevValue) {
+  return value.split(',')
+}
+
+const program = new Command()
+program
+  .usage('')
+  .option('-h, --handles <handles>', 'Audius handles', commaSeparatedList, [])
+  .option('-i, --user-ids <userIds>', 'Audius user ids', commaSeparatedList, [])
+
+const discoveryProviderEndpoint = process.env.DISCOVERY_PROVIDER_ENDPOINT
+
+async function getUserByHandle(handle, discoveryProviderEndpoint) {
+  try {
+    return (
+      await axios({
+        url: `/v1/full/users/handle/${handle}`,
+        method: 'get',
+        baseURL: discoveryProviderEndpoint
+      })
+    ).data.data[0]
+  } catch (err) {
+    throw new Error(
+      `Failed to get creator node endpoint and wallet from endpoint: ${discoveryProviderEndpoint} and handle: ${handle} with ${err}`
+    )
+  }
+}
+
+async function getUserById(userId, discoveryProviderEndpoint) {
+  try {
+    const resp = (
+      await axios({
+        url: `/users?id=${userId}`,
+        method: 'get',
+        baseURL: discoveryProviderEndpoint
+      })
+    ).data.data[0]
+
+    if (!resp) {
+      throw new Error(`Failed to find user with userId ${userId}`)
+    }
+
+    return resp
+  } catch (err) {
+    throw new Error(
+      `Failed to get creator node endpoint and wallet from endpoint: ${discoveryProviderEndpoint} and user id: ${userId} with ${err}`
+    )
+  }
+}
+
+async function getClockValues({
+  wallet,
+  creator_node_endpoint: creatorNodeEndpoint,
+  handle
+}) {
+  const primaryCreatorNode = CreatorNode.getPrimary(creatorNodeEndpoint)
+  const secondaryCreatorNodes = CreatorNode.getSecondaries(creatorNodeEndpoint)
+
+  if (!creatorNodeEndpoint) {
+    return {
+      primaryNode: '',
+      primaryClockValue: '',
+      secondaryNodes: [],
+      secondaryClockValues: [],
+      handle
+    }
+  }
+
+  return {
+    primaryNode: primaryCreatorNode,
+    primaryClockValue: await CreatorNode.getClockValue(
+      primaryCreatorNode,
+      wallet
+    ),
+    secondaryNodes: secondaryCreatorNodes,
+    secondaryClockValues: await Promise.all(
+      secondaryCreatorNodes.map(secondaryNode =>
+        CreatorNode.getClockValue(secondaryNode, wallet)
+      )
+    ),
+    handle
+  }
+}
+
+// get clock values for all users / some users via userIds / handles
+function getUserClockValues(handles, userIds) {
+  const usersFromHandles = handles.map(handle =>
+    getUserByHandle(handle, discoveryProviderEndpoint)
+  )
+
+  const usersFromIds = userIds.map(userId =>
+    getUserById(userId, discoveryProviderEndpoint)
+  )
+
+  return Promise.all(
+    [...usersFromHandles, ...usersFromIds].map(
+      async user => await getClockValues(await user)
+    )
+  )
+}
+
+async function run() {
+  const { handles, userIds } = parseArgsAndEnv()
+  const userClockValues = await getUserClockValues(handles, userIds)
```

Comment:
if the sole purpose of this for loop is to log out `userClockValues`, do you think we can instead add logs in `getUserClockValues()` ?

Reply:
Yes; however, this abstracts out `getUserClockValues()` as something that can be directly called to get all clock values.

This way, in the future, where we might not need to log the output, we can directly call the underlying function.